src/ai/defaultai.cc

```diff
@@ -2976,21 +2976,18 @@ bool DefaultAI::construct_building(const Time& gametime) {
 					number_of_same_nearby += same_it->second;
 				}
 
-				// Considering distance to wh for productionsites:
-				// All productionsites in one place
-
-				if (bo.is(BuildingAttribute::kWell) && bo.is(BuildingAttribute::kRanger) &&
+				// Considering distance to wh for various types of productionsites:
+				if (bo.is(BuildingAttribute::kWell) || bo.is(BuildingAttribute::kRanger) ||
 				    bo.is(BuildingAttribute::kHunter)) {
-					prio += wh_distance_malus;
-				} else if (bo.is(
-				              BuildingAttribute::
-				                 kSpaceConsumer)) {  // ??? &&
-					                                  // bo.is(BuildingAttribute::kSupportingProducer))
-					                                  // {
-					prio += wh_distance_malus;
-				} else if (!bo.inputs.empty() && !bo.ware_outputs.empty() &&
-				           !bo.is(BuildingAttribute::kSupportingProducer)) {
-					prio -= wh_distance_malus;  // push them closer from nearest wh
+					prio += wh_distance_malus;  // push farer
+				} else if (bo.is(BuildingAttribute::kRecruitment)) {
+					prio -= wh_distance_malus;  // push closer
+				} else if (bo.is(BuildingAttribute::kSupportingProducer)) {
```

Comment:
this line is superfluous. this was not what I meant

Reply:
Right now it does what you want and covers also spaceconsumers.

So you say that spaceconsumers should be pushed farer no matter if they are or are not supportingproducers (if there are any such buildings)

Reply:
yes everything alright with Space consumers. 
just delete the part with supporting producers

Reply:
So like this:

```
} else if (bo.is(BuildingAttribute::kRecruitment)) {
	prio -= wh_distance_malus; // push closer
}else if (bo.is(BuildingAttribute::kSpaceConsumer)) {
	prio += wh_distance_malus;  // push farer
} else if (!bo.inputs.empty() && !bo.ware_outputs.empty() && !bo.is(BuildingAttribute::kSupportingProducer)) {
	prio -= wh_distance_malus;  // push closer
```
My objection is that SpaceConsumers that are also SupportingProducers (if any, I do not know) should not be pushed farer / malus should not be applied to them. But if it is fine for you....

Reply:
no like this:
else if (!bo.inputs.empty() && !bo.ware_outputs.empty() && !bo.requires_supporters)) {
	prio -= wh_distance_malus;  // push closer

there are supporting producers but they still need space so should be safe to push them away from warehouses

Reply:
ok